influxql/iterator.gen.go

```diff
@@ -494,6 +494,103 @@ func (itr *floatParallelIterator) monitor() {
 	}
 }
 
+type lazyFloatIterator struct {
+	itr FloatIterator
+	fn  func() (Iterator, error)
+}
+
+// init instantiates the underlying iterator.
+func (itr *lazyFloatIterator) init() error {
+	if itr.itr != nil {
+		return nil
+	}
+
+	i, err := itr.fn()
+	if err != nil {
+		return err
+	} else if i == nil {
+		return nil
+	}
+
+	if it, ok := i.(FloatIterator); ok {
+		itr.itr = it
+	} else {
+		return fmt.Errorf("invalid lazy float iterator type: %T", i)
+	}
+
+	return nil
+}
+
+func (itr *lazyFloatIterator) Close() error {
+	if itr.itr == nil {
+		return nil
+	}
+	return itr.itr.Close()
+}
+
+func (itr *lazyFloatIterator) Next() (*FloatPoint, error) {
+	if err := itr.init(); err != nil {
+		return nil, err
+	} else if itr.itr == nil {
+		return nil, nil
+	}
+	return itr.itr.Next()
+}
+
+func (itr *lazyFloatIterator) Stats() IteratorStats {
```

Comment:
To add a bit to this, I'm uncertain now if the feature ever worked correctly. When I tested this to confirm my suspicions, the limit was still reached. This makes me think that the number of series gets counted as a separate series per shard so if you have 300 series with 2 shards, you end up with 600 series and it would fail if the limit was set to 500.

So if we don't have the original problem anymore, it might just be best to remove that limit.